nexus/types/src/external_api/params.rs

```diff
@@ -83,6 +89,44 @@ pub struct OptionalProjectSelector {
     pub project_selector: Option<ProjectSelector>,
 }
 
+#[derive(Deserialize, JsonSchema)]
+pub struct DiskSelector {
+    #[serde(flatten)]
+    pub project_selector: Option<ProjectSelector>,
+    pub disk: NameOrId,
+}
+
+impl DiskSelector {
+    pub fn new(
+        organization: Option<NameOrId>,
+        project: Option<NameOrId>,
+        disk: NameOrId,
+    ) -> Self {
+        DiskSelector {
+            project_selector: project
+                .map(|p| ProjectSelector::new(organization, p)),
+            disk,
+        }
+    }
+}
+
+#[derive(Deserialize, JsonSchema)]
+pub struct DiskList {
+    #[serde(flatten)]
+    pub project_selector: ProjectSelector,
+    #[serde(flatten)]
+    pub pagination: PaginatedByName,
```

Comment:
as below: I'm not sure how this works.

Reply:
Fixed in [7f608d1](https://github.com/oxidecomputer/omicron/pull/2008/commits/7f608d1d79151efb1ab0c21056f31fd688a92d9c)